larpandoracontent/LArVertex/TrainedVertexSelectionAlgorithm.cc

```diff
@@ -296,8 +296,8 @@ void TrainedVertexSelectionAlgorithm::GetEventShapeFeatures(const ClusterList &c
     }
 
     const float xSpan(this->GetCoordinateSpan(xMax, xMin));
-    const float ySpan(this->GetCoordinateSpan(yMax, zMin));
-    const float zSpan(this->GetCoordinateSpan(yMax, zMin));
+    const float ySpan(this->GetCoordinateSpan(yMax, yMin));
+    const float zSpan(this->GetCoordinateSpan(zMax, zMin));
 
     // Calculate the volume and longitudinality of the event (ySpan often 0 - to be investigated).
     if ((xSpan > std::numeric_limits<float>::epsilon()) && (ySpan > std::numeric_limits<float>::epsilon()))
```

Comment:
Will now never evaluate as true (or not unless algorithm is retasked to work with 3D hit inputs; unlikely). 

Similarly, line 303 will never evaluate as true.

This then leaves line 315, which should nearly always evaluate as true. In this case, we probably want to modify `eventVolume` to `eventArea = xSpan * zSpan` and `longitudinality = zSpan / (xSpan + zSpan)`?

If `xSpan` and `zSpan` are zero, should leave the event feature uninitialised, so it should skip over in MVA.

Reply:
Thanks for the comments @PandoraPFA, I'm currently updating the branch to reflect this. I agree that `eventVolume` should become `eventArea`. However, for `longitudinality` I would imagine we want it to be `zSpan / (xSpan + xSpan)` ? Or even potentially `zSpan / xSpan` for the 2D approach?

Reply:
Hi @henrylay97 - I think we now have the opportunity to redevelop the event shapes as works best under new training and testing. In terms of my suggestion above, I was (without any deep thought) going with the idea of something that would range from ~0 (when the zSpan is significantly smaller than the xSpan), to 0.5 (for square aspect events) to ~1 (when the zSpan is significantly greater than the xSpan). That might be an OK starting point, but please do feel free to really dig into the role these event shapes are providing and let the final decisions be performance driven. Many thanks!

Reply:
I like John's suggestion here; having a well-defined range is usually a useful thing for machine learning applications, so I imagine it will end up yielding the best performance of those options, it also has the nice property that the only undefined case is for xSpan **and** zSpan equal to zero.

Reply:
I see what you mean. It sounds good to me but I can certainly do some tests to check the performance implications. I've updated the branch with the things we've discussed here but this is mostly for book-keeping. Going through this has raised some points on how we will deal with deprecation for any other changes we potentially make during this study. I intend to train various permutations of the different changes so will include this particular change in those permutations so we can see the effect of that change. Thanks to both of you for your comments!

Reply:
Hi @henrylay97 - a very good comment about how we will deal with deprecation. I would say that you ultimately don't want to be constrained by what's there now, so you should feel free to develop so as to obtain the best performance.

When we review what you find, and if you find any outright problems, then we'll have to discuss whether there's a need to allow new code to run with old training outputs, or whether we force a new training pass for everything. We could even split the algorithm implementation if needed. But keep us posted and we can discuss as things evolve. Thanks!